web/thesauruses/javascript/ECMAScript 2020/queues_stacks.json

```diff
@@ -0,0 +1,392 @@
+      ]
+    },
+    "circular_queue_import_statement": {
+      "name": "Import statement to add simple queue",
+      
+      "not-implemented":"true",
+      "comment":"There is no queue library in JavaScript"
+    },
+    "circular_queue_data_type": {
+      "name": "Queue data type name",
+      "code": [
+        "Array [] or class"
+      ]
+    },
+    "circular_queue_create_statement": {
+      "name": "Create a queue",
+      "code": [
+        "//Using array\nlet queue = ['item1','item2','item3']\n//Using linked list\n data = queue()"
+      ]
+    },
+    "circular_queue_create_copy_statement": {
+      "name": "Create a queue from existing queue",
+      "code": [
+        "//Using slice method\n let copyqueue=queue.slice()\n//Using spread operator\n let copyqueue = [...queue]\n//For linked list(implementation down below)\n copyqueue=prevqueue.copy()"
+      ]
+    },
+    "circular_queue_create_copy_from_list_statement": {
+      "name": "Create a queue from a list of items",
+      "code": [
+        "//Using slice method\n let copyqueue=datalist.slice()\n//Using spread operator\n let copyqueue = [...datalist]\n//Using linked list\n copyqueue = queue(datalist) "
+      ]
+    },
+    "circular_queue_destroy_statement": {
+      "name": "Destroy/delete a queue",
+      "code": [
+        "queue = [] \n//Using linked list(implementation below) dataqueue.delete()"
+      ]
+    },
+    "circular_queue_data_structure": {
+      "name": "Data structure that backs the simple queue",
+      "code": [
+        "//Using array \nqueue = [] \n//Using linked list\nclass node\n {\n constructor(data){\n this.data = data;\n this.next = null;\n }\n }\n class queue\n {\n constructor(data=[])\n {\n this.front=null;\n this.rear=null;\n for(let i of data){\n this.push(i)\n }\n }\n copy(){\n let data=[]\n if (this.front == null)return null;\n let temp=this.front;\n while(temp!=null){\n data.push(temp.data)\n temp=temp.next\n }\n return new queue(data)\n }\n delete(){\n this.front=null\n this.rear=null\n }\n push(n){\n let temp =new node(n);\n if(this.front == null){\n this.front = temp;\n }\n else{\n this.rear.next= temp;\n }\n this.rear = temp;\n this.rear.next = this.front\n }\n display(){\n console.log(this.front.data)\n let temp = this.front.next;\n front){\n console.log(temp.data);\n temp = temp.next;\n }\n }\n pop(){\n this.front=this.front.next;\n }\n }\n"
+      ]
+    },
+    
+    "not-implementedrity_queue_import_statement": {
```

Comment:
I think some of the ID fields (like "priority_queue_import_statement") might have had a find-and-replace gone wrong. You can't rename the field names, so some of them (like this one) have "not-implemented" added into the ID. You'll have to undo those, otherwise they'll all show up as "Unknown" on the site.

(I'm guessing you didn't test this first?)